theory.MD

```diff
@@ -0,0 +1,71 @@
+# <p style="text-align: center;">Week 4: Homework</p>
+
+# TASK 1 (Making Your Answer File - Git) :
+>
+> Complete 
+>
+
+# TASK 2 (Theory Questions) :
+
+**Question 1:** 
+>
+> Git is an example of a version control system.
+>
+
+**Question 2:** 
+>
+> Github is a code hosting service, provided by Git.
+> 
+
+**Question 3:** 
+>
+> Modified: This happens when the file has been changed, but the changes are yet to be committed to the repo.
+>
+
+>
+> Staged: This is when the modified file is ready to be commited to the repo.
+>
+
+> 
+> Committed: This is when the changes to a staged file has been saved to the repo. 
+>
+
+**Question 4:** 
+>
+> PR - Pull Request
+>
+
+>
+>Repo - Repository
+>
+
+**Question 5:** 
+>
+> 'git pull'
+>
+
+**Question 6:**
+>
+> 'git fetch'
+>
+
+**Question 7:** 
+>
+> 'git push'
+>
+
+**Question 8:** 
+>
+> The git commit command usually need a little message that you can attach to file before it is commited. This is short, precise message detailing the change that you've commited to help remind you, or fellow coders assessing the code what has changed. You can enter the line - git add, and then - git commit.
+>
+
+# TASK 3 (Exception Handling):
+>
+> Complete
+>
+
+# TASK 4 (Testing): Incomplete 
+>
+> Issues arisen with lack of practice in testing, the terminal won't run the test. I have made sure to use git ls to check the directories, used cat test_shop.py to call on the test_shop.py file, to ensure that gitpod is using the correct file to test. I've also made sure to check if Unittest is installed - which it is. I have also made sure that the file was saved before trying to run any tests. I am also not sure if I satisfied the marking criteria by using Gitpod to run the tests, but my Pycharm has been deactivated, and I do not have the necessary skills to run anything using VSCode until further notice. 
+>
+
```

Comment:
23/40 => 58%, This would have been better if you added more test cases and met the criteria of the question